tests/ui/test_ui.py

```diff
@@ -109,14 +110,21 @@ def test_set_footer_text_with_duration(
         duration: Optional[float] = 5.3,
     ) -> None:
         mocker.patch(VIEW + ".get_random_help", return_value=["some help text"])
-        mock_sleep = mocker.patch("time.sleep")
+        mocked_alarm = mocker.patch(CONTROLLER + ".loop.set_alarm_in")
 
-        view.set_footer_text([custom_text], duration=duration)
+        with freeze_time() as frozen_datetime:
+            view.set_footer_text([custom_text], duration=duration)
 
-        view.frame.footer.set_text.assert_has_calls(
-            [mocker.call([custom_text]), mocker.call(["some help text"])]
-        )
-        mock_sleep.assert_called_once_with(duration)
+            view.frame.footer.set_text.assert_called_once_with([custom_text])
+            assert view.controller.update_screen.call_count == 1
+
+            frozen_datetime.tick(duration)
+
+            # Trigger the alarm
+            alarm_callback = mocked_alarm.call_args[0][1]
+            alarm_callback(None, None)
```

Comment:
If I understand correctly, line 123 is simulating the alarm going off (?) and triggering the callback function. (Correct me if I'm wrong)

Could we make the comment on that line more specific about this to prevent any confusion? We could start the comment with `# NOTE:` similar to how it's done elsewhere.

Reply:
Yes, the comment is meant to convey that the code below is to trigger the alarm, i.e directly call the callback function. I wasn't completely sure if it's required, just something that I would personally find convenient, if I was reading the code.

Ok, I'll clarify that, thank you for the feedback. Would "Directly trigger the alarm's callback function" be a better description?

Ah, we do seem to have a few comments with `#NOTE:` in that file, but generally in the codebase, we don't use it that much. Not sure what the convention is regarding its usage in our repo. There doesn't seem to be a global convention. So, I'll wait to hear from the others regarding that.